crates/api_common/src/utils.rs

```diff
@@ -505,13 +506,99 @@ pub async fn check_private_instance_and_federation_enabled(
   Ok(())
 }
 
-pub async fn remove_user_data(banned_person_id: PersonId, pool: &DbPool) -> Result<(), LemmyError> {
+pub async fn purge_image_posts_for_person(
```

Comment:
This file is already quite big, so please put all the purge/remove related stuff into a new file.

Reply:
Can do that in a separate PR, I'll make an issue for it. Utils touches almost every file so that'd make this PR huge.